foundry.md

```diff
@@ -383,6 +385,97 @@ This rule takes `uint256` value using `#asWord(#range(LM, ARGSTART +Int 4, 32)`
+       andBool #asWord(#range(LM, ARGSTART, 4)) ==Int 1719639408 // selector ( "load(address,bytes32)" )
+      [priority(40)]
+
+
+    syntax KItem ::= "#foundryVmLoad" Int Int Int
+ // ----------------------------------------------------------
+    rule <k> #foundryVmLoad ARGSTART RETSTART RETWIDTH
+          => #setLocalMem RETSTART RETWIDTH #bufStrict(32, #lookup(STORAGE, #asWord(#range(LM, ARGSTART +Int 36, 32))))
+          ~> 1 ~> #push ...
+         </k>
+         <output>
+           _ => #bufStrict(32, #lookup(STORAGE, #asWord(#range(LM, ARGSTART +Int 36, 32))))
+         </output>
+         <account>
+             <acctID> ACCTID </acctID>
+			 <storage> STORAGE </storage>
+             ...
+         </account>
+         <localMem> LM </localMem>
+      requires ACCTID ==Int #asWord(#range(LM, ARGSTART +Int 4, 32))
+```
+
+#### `store` - Stores a value to an address' storage slot.
+
+```
+function store(address account, bytes32 slot, bytes32 value) external;
+```
+
+`call.store` will match when the `store` function is called at the [Foundry cheatcode address](https://book.getfoundry.sh/cheatcodes/#cheatcodes-reference).
+This rule then takes the account using `#asWord(#range(LM, ARGSTART +Int 4, 32))` and the new slot value using `#asWord(#range(LM, ARGSTART +Int 68, 32))` and updates the slot denoted by `#asWord(#range(LM, ARGSTART +Int 36, 32))`.
+
+```k
+    rule [call.store]:
+         <k> CALL _ CHEAT_ADDR _VALUE ARGSTART _ARGWIDTH _RETSTART _RETWIDTH
+          => #loadAccount #asWord(#range(LM, ARGSTART +Int 4, 32))
+          ~> #foundryVmStore ARGSTART ...
+         </k>
+         <localMem> LM </localMem>
+      requires CHEAT_ADDR ==Int #address(FoundryCheat)
+       andBool #asWord(#range(LM, ARGSTART, 4)) ==Int 1892290747 // selector ( "store(address,bytes32,bytes32)" )
+      [priority(40)]
+
+
+    syntax KItem ::= "#foundryVmStore" Int
+ // ----------------------------------------------------------
+    rule <k> #foundryVmStore ARGSTART => 1 ~> #push ... </k>
+         <output> _ => .ByteArray </output>
+         <account>
+             <acctID> ACCTID </acctID>
+			 <storage> STORAGE => STORAGE [ #asWord(#range(LM, ARGSTART +Int 36, 32))
+                                            <- #asWord(#range(LM, ARGSTART +Int 68, 32)) ] </storage>
+             ...
+         </account>
+         <localMem> LM </localMem>
+      requires ACCTID ==Int #asWord(#range(LM, ARGSTART +Int 4, 32))
+```
+
+Utils
+-----
+
+- `#loadAccount ACCT` creates a new, empty account for `ACCT` if it does not already exist. Otherwise, it has no effect
+
+```k
+    syntax KItem ::= "#loadAccount" Int
+ // ---------------------------------
+    rule <k> #loadAccount ACCT => . ... </k>
+         <activeAccounts> ACCTS:Set </activeAccounts>
+      requires ACCT in ACCTS
+
+    rule <k> #loadAccount ACCT => #newAccount ACCT ... </k>
+         <activeAccounts> ACCTS:Set </activeAccounts>
+      requires notBool ACCT in ACCTS
+```
+
```

Comment:
Can we use the existing `#newAccount` instead? Or have this desugar to `#newAccount` somehow? It's defined in `evm.md`.

Reply:
The second rule already desugars to `#newAccount`. The difference between the two is that `#newAccount` is only defined if the account does not already exist, while `#loadAccount` simply has no effect in this case (see first rule above). I think this works well in the context of most cheatcodes, which don't care if the specified account already exists or has to be created, but I'm open for suggestions.

Reply:
Makes sense! thank you.